src/pysoem/pysoem.pyx

```diff
@@ -239,6 +241,13 @@ cdef class CdefMaster:
             ret_val = cpysoem.ecx_init_redundant(&self._ecx_contextt, &self._ecx_redport, ifname.encode('utf8'), ifname_red.encode('utf8'))
         if ret_val == 0:
             raise ConnectionError('could not open interface {}'.format(ifname))
+
+        self.context_initialized = True
+
+    def __check_context_is_initialized(self):
+        if not self.context_initialized:
+            raise UninitializedContextError("SOEM Context is not initialized or has been closed. Call Master.open() first")
```

Comment:
I'm still not fully convinced by the exception name. The concept fo the context is a SOEM internal concept that pysoem hides form the user. Because the exception type is part of API, the name should not be about the internal concept.

PySerial does a good job on his
```py
import serial

with serial.Serial("COM1") as s:
    pass

s.read()
```
PySerial throws this: `serial.serialutil.PortNotOpenError: Attempting to use a port that is not open`.

What about adopting this and naming the exception `InterfaceNotOpenError("..")` and the message `"Attempting to use an interface that is not open"`?
Instead of "Interface", other names could be "NetworkInterface" or "Adapter".